Improve tabular datatypes.

jmchilton opened the pull request:
This PR merges the stale PR #1156 into the latest dev branch and introduces a hack into sniffing to fix #1818.

Commit:
Alternative stricter csv formats

Commit:
Flake8 fix

Commit:
Remove bug log statement

jmchilton committed:
Small tweaks to CSV fixes by @Christian-B.

 - Update sample datatype conf.
 - Use Python naming conventions and PEP-8 style fixes.

Christian-B committed:
Merge pull request #1 from jmchilton/csv_fix

Small tweaks to CSV fixes by @Christian-B.

Commit:
fixed spelling found by @nsoranzo

jmchilton committed:
Merge remote-tracking branch 'jmchilton/dev' into csv_fix

jmchilton committed:
Add test for #1818 to PR #1156.

jmchilton committed:
Merge PR #1156 into dev and fix #1818.

nsoranzo commented:
@jmchilton The original bug description mentioned that "If backward compatibility is not required I will merge the new Excel_CSV datatype into the CSV datatype." What do you think about that?

jmchilton commented:
@nsoranzo I'm indifferent on that point - I think the old datatype was probably new enough and broken enough that we shouldn't assume anyone sub-classed it - but I'm not sure I want to remove it as part of this PR either.

nsoranzo commented on config/datatypes_conf.xml.sample:
```diff
@@ -52,9 +52,8 @@
-    <datatype extension="csv" type="galaxy.datatypes.tabular:CSV" display_in_upload="true" />
-    <!-- End MSI added Datatypes -->
+    <datatype extension="csv" type="galaxy.datatypes.tabular:ExcelCSV" display_in_upload="true" />
+    <datatype extension="tsv" type="galaxy.datatypes.tabular:ExcelCSV" display_in_upload="true" />
```
s/ExcelCSV/ExcelTSV/

nsoranzo commented:
@jmchilton I was thinking something like:
1. overwrite `CSV.sniff()` and `CSV.set_meta()` methods with the ones from `BaseCSV`
2. remove `BaseCSV` class and rename `CSV` to `BaseCSV`
3. rename `ExcelCSV` class to `CSV`

Does this make sense?

martenson commented:
This might be the last blocking issue/PR for the 16.04 freeze - please consider pushing it back to 16.07.
@jmchilton

jmchilton commented:
Sorry for holding up the release - I've changed the milestone - it fixes a bug so I can just open it against 16.04 anyway.

jmchilton committed:
More fixups for specialized, more correct tabular datatypes.

Followed through with @nsoranzo's comment:

1. overwrite CSV.sniff() and CSV.set_meta() methods with the ones from BaseCSV
2. remove BaseCSV class and rename CSV to BaseCSV
3. rename ExcelCSV class to CSV

Also renamed ExcelTSV to TSV and corrected another bug in sample datatypes noticed by @nsoranzo.

nsoranzo commented:
:+1:

martenson commented:
yay! thanks a lot @jmchilton & @nsoranzo & @Christian-B

martenson commented:
this PR made it to freeze, changing its milestone
